sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java

```diff
@@ -92,52 +58,85 @@ public CommunicationTokenCredential(TokenRefresher tokenRefresher) {
 
     /**
      * Create with serialized JWT token and a token supplier to auto-refresh the
-     * token before it expires. Callback function tokenRefresher will be called ahead
-     * of the token expiry by the number of minutes specified by
+     * token before it expires. Callback function tokenRefresher will be called
+     * ahead of the token expiry by the number of minutes specified by
      * CallbackOffsetMinutes defaulted to two minutes. To modify this default, call
      * setCallbackOffsetMinutes after construction
      * 
      * @param tokenRefresher implementation to supply fresh token when reqested
      * @param initialToken serialized JWT token
-     * @param refreshProactively when set to true, turn on proactive fetching to
-     *                           call tokenRefresher before token expiry by minutes
-     *                           set with setCallbackOffsetMinutes or default value
-     *                           of two minutes
+     * @param refreshProactively when set to true, turn on proactive fetching to call
+     *                           tokenRefresher before token expiry by minutes set
+     *                           with setCallbackOffsetMinutes or default value of
+     *                           two minutes
      */
-    public CommunicationTokenCredential(
-        TokenRefresher tokenRefresher, 
-        String initialToken,
-        boolean refreshProactively) 
-    {
+    public CommunicationTokenCredential(TokenRefresher tokenRefresher, String initialToken,
+            boolean refreshProactively) {
         this(tokenRefresher);
         Objects.requireNonNull(initialToken, "'initialToken' cannot be null.");
         setToken(initialToken);
-        tokenFuture = new TokenImmediate(accessToken);
         if (refreshProactively) {
             OffsetDateTime nextFetchTime = accessToken.getExpiresAt().minusMinutes(DEFAULT_EXPIRING_OFFSET_MINUTES);
             fetchingTask = new FetchingTask(this, nextFetchTime);
         }
     }
 
+    private class TokenNext extends Mono<AccessToken> {
```

Comment:
This class name reads a bit funny to me. What about TokenAccessor? English is my second language though. :-)

Reply:
Updated to TokenRetriever, hope that makes sense too